File: data-r2dbc-entitytemplate/src/main/java/com/example/demo/Post.java
package com.example.demo;

import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Version;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.util.List;
import java.util.UUID;

@Table(value = "posts")
record Post(

        @Id
        @Column("id")
        UUID id,

        @Column("title")
        String title,

        @Column("content")
        String content,

        @Column("status")
        Status status,

        @Column("tags")
        List<String> tags,

        @Column("version")
        @Version
        Long version
) {

    public static Post of(String title, String content) {
        return new Post(null, title, content, null, List.of("default"), null);
    }

    public static Post of(String title, String content, List<String> tags) {
        return new Post(null, title, content, null, tags, null);
    }

    enum Status {
        DRAFT, PENDING_MODERATION, PUBLISHED;
    }

}

File: boot-jooq-kotlin-co-gradle/src/main/kotlin/com/example/demo/domain/model/Post.kt
package com.example.demo.domain.model

import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import java.time.LocalDateTime
import java.util.*

@Table(value = "posts")
data class Post(
    @Id
    @Column("id")
    val id: UUID? = null,

    @Column("title")
    var title: String? = null,

    @Column("content")
    var content: String? = null,

    @Column("tags")
    var tags: List<String>? = emptyList(),

    @Column("status")
    var status: Status? = Status.DRAFT,

    @Column("created_at")
    @CreatedDate
    val createdAt: LocalDateTime? = null,

    @Column("comments_count")
    var commentsCount: Long? = 0,

//    @Column("version")
//    @Version
//    val version: Long? = null,
) {
    enum class Status {
        DRAFT, PENDING_MODERATION, PUBLISHED
    }
}